### PR TITLE
Layer attribute search control

### PR DESCRIPTION
**Pull request by sackh**
Hi @martinRenou 
I have created a new PR for issue #585  as the previous one got closed by mistake.
Please review this.

### CODE REVIEW

File: js/src/controls/SearchControl.js [hunk 1]
```diff
@@ -19,22 +19,43 @@ export class LeafletSearchControlModel extends control.LeafletControlModel {
...
-      options.marker = view.obj;
-      this.obj = L.control.search(options);
-    });
+    if (this.model.get('layer') !== null) {
+        return this.create_child_view(this.model.get('layer')).then((layer_view) => {
+            let options = this.get_options();
+            options.layer = layer_view.obj;
+            options.marker = false;
+            this.obj = L.control.search(options);
+        });
```
Comment (martinRenou) [inline js/src/controls/SearchControl.js:42]: It looks like your implementation makes it impossible to set the `marker` and the `layer` attributes at the same time.

What about something like this for the `create_obj` method:
```javascript
const layer = this.model.get('layer');
const marker = this.model.get('marker');

const layer_promise = layer !== null ? this.create_child_view(layer) : Promise.resolve(null);
const marker_promise = marker !== null ? this.create_child_view(marker) : Promise.resolve(null);

return Promise.all([layer_promise, marker_promise]).then((layer_view, marker_view) => {
    const options = this.get_options();
    options.layer = layer_view !== null ? layer_view.obj : false;
    options.marker = marker_view !== null ? marker_view.obj : false;
    this.obj = L.control.search(options);
});
```

I did not try it locally so you might need to change one or two things.

File: js/src/controls/SearchControl.js [hunk 2]
```diff
@@ -19,22 +19,43 @@ export class LeafletSearchControlModel extends control.LeafletControlModel {
...
+            options.marker = view.obj;
+            this.obj = L.control.search(options);
+        });
+    }
+  }
+
+  leaflet_events() {
+    if (this.model.get('layer') !== null) {
+        this.obj.on('search:locationfound', function(e) {
+            e.layer.setStyle({fillColor: '#3f0', color: '#0f0'});
```
Comment (martinRenou) [inline js/src/controls/SearchControl.js:55]: Should we remove this style once the search control searches for something else?

Comment (sackh) [reply js/src/controls/SearchControl.js:55]: Currently, once you clicked on the searched feature style gets removed automatically. I will have to check if there is any other event to clear the style.

File: js/src/controls/SearchControl.js (hunk 1 above)
Comment (sackh) [reply js/src/controls/SearchControl.js:42]: I will try this.

Comment (martinRenou) [reply js/src/controls/SearchControl.js:42]: You could indeed do that if the user did not define a Marker? But if the user defined a Marker I guess they will expect it to be used.

Comment (sackh) [reply js/src/controls/SearchControl.js:42]: Sure, I will try the changes you suggested.

Comment (sackh) [reply js/src/controls/SearchControl.js]: @martinRenou - updated the code as suggested by you.

**Comment by martinRenou** [comment]
We might want to make the "found" style an attribute, but let's do it later :) Thanks a lot for your work!

**Pull request merged**